### ISSUE TITLE
add datasource to pull all available log fields

### ISSUE THREAD

**Issue by vital4ik** [cloudflare/terraform-provider-cloudflare#1005]
### Current Terraform version

0.14.8

### Description

is it possible to add a datasource to pull all the log fields? 
they are available here:
https://api.cloudflare.com/client/v4/zones/<zone_id>/logpush/datasets/<dataset>/fields. 

### Use cases

in order to properly construct the logpush job that is up to date with the fields available, would be nice to have it as a datasource

### Potential Terraform configuration

none

**Comment by vitdolphin** [comment]
hi @jacobbednarz, which upstream library needs to be updated for this?
using null provisioner with local exec to get this is very fragile

**Comment by jacobbednarz** [comment]
the terraform provider uses [cloudflare-go](https://github.com/cloudflare/cloudflare-go) under the hood for the API calls; it doesn't implement anything itself. so before we can do anything here, that library needs it.

**Comment by tjstansell** [comment]
cloudflare-go should support the required api calls now that it supports account-level logpush jobs via `GetAccountLogpushFields` and `GetZoneLogpushFields`.  I'll probably try to get around to working on a PR this week unless someone else works on it.

**Comment by tjstansell** [comment]
fwiw, i've also requested that logpush/logpull simply support `fields=ALL` or similar to remove this requirement entirely when you just want everything... but who knows if/when that'll ever get implemented.

**Comment by jacobbednarz** [comment]
This issue has been closed as we are now tracking this internally with service teams directly. If you would like an update or to be notified when/if the product ships with this change, please reach out to Cloudflare Support or your account team who can watch the internal feature request for you.
